### PR TITLE
Add hevc support cpu frames decoder

### PR DESCRIPTION
**Pull request by awolant**
## Category:

New feature

## Description:

Add basic HEVC support for `FramesDecoder` on CPU

### Affected modules and functionalities:

`FramesDecoder` and its tests

### Key points relevant for the review:

Is it tested properly?

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [x] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [x] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2688

### CODE REVIEW

File: dali/operators/reader/loader/video/frames_decoder.cc [hunk 1]
```diff
@@ -51,6 +56,15 @@ void FramesDecoder::InitAvState() {
   DALI_ENFORCE(av_state_->packet_, "Could not allocate av packet");
 }
 
+bool FramesDecoder::CheckCodecSupport() {
```
Comment (banasraf) [inline dali/operators/reader/loader/video/frames_decoder.cc:59]:
```c++
return std::find(SupportedCodecs.begin(), SupportedCodecs.end(), av_state_->codec_->id) != SupportedCodecs.end();
```
nitpick, up to you

File: dali/operators/reader/loader/video/frames_decoder_test.cc [hunk 2]
```diff
@@ -184,6 +194,18 @@ TEST_F(FramesDecoderTest_CpuOnlyTests, InvalidSeek) {
...
+TEST_F(FramesDecoderTest_CpuOnlyTests, InvalidCodec) {
+  std::string path = testing::dali_extra_path() + "/db/video/vp9/vp9_0.mp4";
+
+  try {
+    FramesDecoder file(path);
+  } catch (const DALIException &e) {
+    EXPECT_TRUE(
+      strstr(e.what(),
+      make_string("Unsupported video codec: vp9 in file: ", path).c_str()));
+  }
```
Comment (banasraf) [inline dali/operators/reader/loader/video/frames_decoder_test.cc:206]: This test will pass if no exception is thrown, right? Do we want to fail in such case?

**Comment by awolant** [comment]
!build

File: dali/operators/reader/loader/video/frames_decoder_test.cc (hunk 2 above)
Comment (awolant) [reply dali/operators/reader/loader/video/frames_decoder_test.cc]: Nicely caught. I extracted some helper functions to run these testes where we expect `FramesDecoder` to throw and refactored all of those tests that do that. In the helper I added a `FAIL()`, if we do not throw.

File: dali/operators/reader/loader/video/frames_decoder_test.cc [hunk 3]
```diff
@@ -92,6 +92,17 @@ class FramesDecoderTestBase : public VideoTestBase {
...
+  void RunFailureTest(std::function<void()> body, std::string expected_error) {
+    try {
+      body();
+
+      FAIL();   // If we reached this point body did not throw an exception.
+    } catch (const DALIException &e) {
+      EXPECT_TRUE(
+        strstr(e.what(), expected_error.c_str()));
+    }
+  }
```
Comment (awolant) [inline dali/operators/reader/loader/video/frames_decoder_test.cc:104]: Helper function to run failure tests.

File: dali/operators/reader/loader/video/frames_decoder_test.cc [hunk 4]
```diff
@@ -111,6 +122,12 @@ class FramesDecoderTest_CpuOnlyTests : public FramesDecoderTestBase {
     return frame_buffer_.data();
   }
 
+  void RunConstructorFailureTest(std::string path, std::string expected_error) {
+    RunFailureTest([&]() -> void {
+      FramesDecoder decoder(path);},
+      expected_error);
+  }
```
Comment (awolant) [inline dali/operators/reader/loader/video/frames_decoder_test.cc:129]: Helper function to run `FramesDecoder` constructor failure tests.

**Comment by awolant** [comment]
!build

**Review by szalpal — APPROVED** [review]

**Pull request merged**